nixos/modules/services/web-apps/honk.nix

```diff
@@ -0,0 +1,153 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+let
+  cfg = config.services.honk;
+
+  honk-initdb-script = cfg: pkgs.writeShellApplication {
```

Comment:
Why take this extra parameter? It will be inherited from the line above anyway and this script is only used internally.

Reply:
Because I prefer to avoid function to read parameters outside their scope. For sure, it could be done differently, but I believe that specifying each parameter that the function needs to work properly is a better pattern than expecting the function to have access to variables outside its scope.